src/edu/cmu/tartan/Game.java

```diff
@@ -564,9 +611,200 @@ private void help() {
+            for(Item i : r.items){
+                itemList.add(i.toString());
+            }
+            room.put("room",r.shortDescription());
+            room.put("items",itemList);
+            roomList.put("room"+roomArrayList.indexOf(r),room);
+        }
+        return roomList;
+    }
+
+    public JSONArray visitedRoomListConvertJSONArray(){
+        //save vistied room list
+        JSONArray visitedRooms = new JSONArray();
+        for(Room r : player.getRoomsVisited()){
+            visitedRooms.add(r.shortDescription());
+        }
+        return visitedRooms;
+    }
+
+    public JSONArray collectedItemListConvertJSONArray(){
+        //save player's item list
+        JSONArray collectedItems = new JSONArray();
+        for(Item i : player.getCollectedItems()){
+            collectedItems.add(i.toString());
+        }
+        return collectedItems;
+    }
+
+    public void save(int period){
+		SaveTask saveTask = new SaveTask(this);
+		saveScheduler.schedule(saveTask,100,period);
+	}
+
+	public synchronized void save()  {
+		printOutInterface.console("Save start");
+
+		JSONObject jsonObject = new JSONObject();
+
+		//Save this.roomArrayList
+		jsonObject.put("roomList",roomArrayListConvertJSONObject());
+
+        //Save player.getRoomsVisited()
+		jsonObject.put("visitedRoomList",visitedRoomListConvertJSONArray());
+
+		//Save player score
+		jsonObject.put("score",player.getScore());
+
+        //Save player.getCollectedItems()
+		jsonObject.put("collectedItems",collectedItemListConvertJSONArray());
+
+		//Save player current room
+		jsonObject.put("currentRoom",player.currentRoom().shortDescription());
+
+        try {
+            writeFileInterface.write(gameName,jsonObject.toJSONString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        printOutInterface.console("Save Done");
+	}
+
+	public List<Room> loadRoomArrayList(JSONObject roomList){
+        //make load rooms status
+        List<Room> loadedRooms = new ArrayList<>();
+
+        for(int i = 0; i<roomList.size(); i++){
+
+            JSONObject roomNumber = (JSONObject) roomList.get("room"+i);
+
+            //get room description and make room
+            String roomDescription = (String) roomNumber.get("room");
+            Room room = new Room("",roomDescription);
+
+            //make item list
+            JSONArray itemList = (JSONArray) roomNumber.get("items");
+            ArrayList<Item> itemArrayList = new ArrayList<>();
+            Iterator<String> iterator = itemList.iterator();
+            while (iterator.hasNext()){
+                itemArrayList.add(Item.getInstance(iterator.next()));
+            }
+
+            //clear item list then put new item list
+            room.items.clear();
+            room.putItems(itemArrayList);
+
+            loadedRooms.add(room);
+        }
+
+        return loadedRooms;
+    }
+
+    public void updateRoomsItemList(List<Room> loadedRooms){
```

Comment:
Storing room array index would be better for restoring room information than using double 'for' statement.